src/Controller/StationsController.php

```diff
@@ -182,11 +282,138 @@ public function stationPage(Request $request, string $slug): Response
             'breadcrumbs' => $this->breadcrumbsGenerator->getBreadcrumbs($request->getPathInfo(), $activePageBreadCrumb),
             'individualForm' => $individualForm->createView(),
             'observationForm' => $observationForm->createView(),
         ]);
     }
+
+    /**
+     * @Route("/station/{stationId}/delete", name="station_delete")
+     */
+    public function stationDelete(Request $request, int $stationId): Response
+    {
+        /**
+         * @var Station $station;
+         */
+        $station = $this->deleteEntityObject(Station::class, $stationId);
+
+        if (!$station) {
+            throw $this->createNotFoundException('La station n’existe pas');
+        }
+
+        $individuals = $this->getDoctrine()
+            ->getRepository(Individual::class)
+            ->findSpeciesIndividualsForStation($station)
+        ;
+
+        foreach ($individuals as $individual) {
+            $this->deleteEntityObject(Individual::class, $individual->getId());
+        }
+
+        $this->addFlash('notice', 'La station a été supprimée');
+
+        return $this->redirectToRoute('stations');
+    }
+
+    /**
+     * @Route("/individual/{individualId}/delete", name="individual_delete")
+     */
+    public function individualDelete(Request $request, int $individualId): Response
+    {
+        /**
+         * @var Individual $individual;
+         */
+        $individual = $this->deleteEntityObject(Individual::class, $individualId);
+
+        if (!$individual) {
+            throw $this->createNotFoundException('L’individu n’existe pas');
+        }
+        $observations = $this->getDoctrine()->getRepository(Observation::class)
+            ->findBy(['individual' => $individual], ['date' => 'DESC'])
+        ;
+
+        foreach ($observations as $observation) {
+            /*
+             * @var Observation $observation;
+             */
+            $this->deleteEntityObject(Observation::class, $observation->getId());
+        }
+
+        $this->addFlash('notice', 'L’individu a été supprimé');
+
+        return $this->redirectToRoute('stations_show', [
+            'slug' => $individual->getStation()->getSlug(),
+        ]);
+    }
+
+    /**
+     * @Route("/observation/{obsId}/delete", name="observation_delete")
+     */
+    public function observationDelete(Request $request, int $obsId): Response
+    {
+        /**
+         * @var Observation $obs;
+         */
+        $obs = $this->deleteEntityObject(Observation::class, $obsId);
+
+        if (!$obs) {
+            throw $this->createNotFoundException('Cette observation n’existe pas');
+        }
+
+        $this->addFlash('notice', 'L’observation a été supprimée');
+
+        return $this->redirectToRoute('stations_show', [
+            'slug' => $obs->getIndividual()->getStation()->getSlug(),
+        ]);
+    }
+
+    /**
+     * @return Station|Observation|Individual|null
+     */
+    private function deleteEntityObject(string $entity, int $id)
```

Comment:
Don't implement method to delete entities, just use `$manager->remove($observation);`